### PR TITLE
Add file path as callback parameter

### PR DESCRIPTION
**Pull request by mfuentesg**
The matched file path it is important information

### CODE REVIEW

**Comment by shellscape** [comment]
thanks for the PR. can you provide a use-case for this, and perhaps another gulp plugin you're using which follows the same pattern?

**Comment by mfuentesg** [comment]
I have an application separated in js-modules, each module has a specific version that it is included in a php-file.

But, this modules are versioned and i want to remove it from git.
Many files, has a reference to missing files so i want find the references to non existent files and fix it.

```
Sample Tree
├── modules
  ├── some_module
     └── main_v1.0.1.js  (minified file)
```

``` php
/** Sample import */
$this->superModule([
    'appVersion' => 'modules/some_module/main_v1.0.0', // Missed module
]);
```

``` js
/** Sample Task */
var scan = require('gulp-scan');
var fs = require('fs');
var basePath = __dirname__ . '/modules/';

gulp.task('missed-modules', function() {
  gulp.src('views/**/*.php').pipe(scan({
    term: /modules.*main.*[0-9]/gm,
    fn: function(match, filePath) {
      var file = basePath + match + '.js';

      fs.stat(file, function(err) {
        if (err !== null) {
          console.log('Missed Module: ' + file);
          console.log(filePath);
          console.log('---');
        }
      });
    }
  }));
});
```

**Comment by shellscape** [comment]
solid use case. rather than passing only the filePath to the callback, I'd like to pass a clone of the entire file object. I spoke with one of the gulp maintainers and they suggested that as a best practice. That way all of the file's properties are available to the callback, but there's no risk of modifying the original file in the callback (and avoiding weird bugs).

If you'd like to do that work, I'd appreciate it. If not, no worries and I can get to it later.

**Comment by mfuentesg** [comment]
No problem, i will fix it

**Comment by mfuentesg** [comment]
Done!

File: index.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ module.exports = function (options) {
 			matches = content.match(term);
 
 			if(matches !== null) {
-				matches.forEach(options.fn);
+				matches.forEach(function(match) {
+					options.fn(match, file);
```
Comment (shellscape) [inline index.js]: should be file.clone()

File: specs/specs.js [hunk 2]
```diff
@@ -15,8 +15,9 @@ var expect = require('chai').expect,
 
 describe('Scan Fn', function () {
 	it('should scan using a string', function (done) {
-		var stream = scan({ term: '@import', fn: function (match) {
+		var stream = scan({ term: '@import', fn: function (match, matchedFile) {
 				expect(match).to.equal('@import');
+				expect(matchedFile).to.equal(file);
```
Comment (shellscape) [inline specs/specs.js]: will likely fail after the file.clone change. compare a few properties of the files instead

File: specs/specs.js [hunk 3]
```diff
@@ -27,9 +28,10 @@ describe('Scan Fn', function () {
 	});
 
 	it('should scan using a RegExp', function (done) {
-		var stream = scan({ term: /@import(.+)$/gm, fn: function (match) {
+		var stream = scan({ term: /@import(.+)$/gm, fn: function (match, matchedFile) {
 				expect(match).to.have.length.above(0);
 				expect(match.indexOf('@import')).to.equal(0);
+				expect(matchedFile).to.equal(file);
```
Comment (shellscape) [inline specs/specs.js]: same as above

**Comment by mfuentesg** [comment]
Done again

**Comment by shellscape** [comment]
:+1:  thank you!

**Comment by shellscape** [comment]
0.1.2 published

**Pull request merged**